src/internal/Subscription.ts

```diff
@@ -168,13 +160,21 @@ export class Subscription implements SubscriptionLike {
           subscription._subscriptions = [tmp];
         }
         break;
-      default:
+      default: {
+        if (!(<any>teardown)) {
```

Comment:
Is this just leftover defensive programming? Or is this on purpose?

Reply:
Seems like it's leftover.